### PR TITLE
FLUID-6158: Add fluid.modelPairToChanges()

### PR DESCRIPTION
**Pull request by simonbates**
See: https://issues.fluidproject.org/browse/FLUID-6158

Documentation pull request: https://github.com/fluid-project/infusion-docs/pull/122

### CODE REVIEW

File: src/framework/core/js/DataBinding.js [hunk 1]
```diff
@@ -1507,4 +1507,76 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
...
     };
 
+    /**
+     * Calculates the changes between the model values 'value' and
+     * 'oldValue'. The optional argument 'changePathPrefix' is prepended
+     * to the generated change paths. This is useful for generating
+     * change records to be applied at a non-root path in a model.
+     * Returns an array of change records, suitable for passing to
+     * fluid.fireChanges().
+     */
```
Comment (amb26) [inline src/framework/core/js/DataBinding.js]: Some Javadoc-style parameter comments, as we sometimes see elsewhere,  would be useful here - e.g. @param changePathPrefix {String}

File: src/framework/core/js/DataBinding.js [hunk 2]
```diff
@@ -1507,4 +1507,76 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
...
+            fluid.pathUtil.parseEL(changePathPrefix),
+            diffOptions.changeMap,
+            [],
+            changes
+        );
+
+        return changes;
+    };
+
+    fluid.modelPairToChanges1 = function (value, changePathPrefixSegs, changeMap, changeSegs, changes) {
```
Comment (amb26) [inline src/framework/core/js/DataBinding.js]: The suffix "1" is a little cryptic - perhaps something like modelPathToChangesRecurse or modelPathToChangesImpl? Doc comment also desirable given the long list and complex workflow

File: src/framework/core/js/DataBinding.js [hunk 3]
```diff
@@ -1507,4 +1507,76 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
...
+                    });
+                } else if (change === "DELETE") {
+                    changes.push({
+                        path: changePathPrefixSegs.concat(currentChangeSegs),
+                        value: null,
+                        type: "DELETE"
+                    });
+                } else if (fluid.isPlainObject(change, true)) {
+                    // Recurse down the tree of changes
+                    fluid.modelPairToChanges1(
```
Comment (amb26) [inline src/framework/core/js/DataBinding.js]: We could probably get a few of these arguments onto the same line? I'm fully supportive of our new shorter line length but still keen to see vertical space used as sparingly as we can given that :)

**Comment by simonbates** [comment]
@amb26 I've address the review comments -- thanks!

**Pull request merged**